lib/customer.rb

```diff
@@ -0,0 +1,48 @@
+require 'csv'
+
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = {
+      street: address[:street],
+      city: address[:city],
+      state: address[:state],
+      zip: address[:zip]
+    }
+  end
+  
+  #returns a collection of Customer instances, representing all of the Customers described in the CSV file
+  def self.all 
+    customer_array = []
+    filename = "/Users/emilyvomacka/Documents/Ada/week_three/grocery-store/data/customers.csv"
```

Comment:
You use the absolute path here, which causes this to crash on my machine. Don't do that! I don't want to change your code. a fix:
```suggestion
    filename = "./data/customers.csv"
```